File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(string s) {
    int answer = 0;
    vector<string> d = {"zero", "one","two","three","four", "five","six","seven","eight","nine"};
    string tmp = "";
    for(int i = 0; i < s.size(); i++){
        if(isalpha(s[i])){
            tmp += s[i];
            if(find(d.begin(), d.end(),tmp)!=d.end()){
                answer = answer * 10 + find(d.begin(), d.end(), tmp) - d.begin();
                tmp = "";
            }
        }
        else if(isdigit(s[i])){
            answer = answer * 10 + (s[i]-'0');
        }
    }
    return answer;
}

File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.py
def solution(s):
    answer = 0
    dic = ["zero","one","two","three","four","five","six","seven","eight","nine"]
    tmp = ""
    for c in s:
        if not c.isdigit():
            tmp += c
        else:
            answer = answer*10+int(c)
        if tmp in dic:
            answer = answer*10+dic.index(tmp)
            tmp = ""
    return answer